### PR TITLE
chore: cache examples and terraform plugins

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
Using caches always and also for terraform plugins should improve stability and reduces the test runtime of the unit tests by 7 minutes

### CODE REVIEW

File: .github/workflows/examples.yml [hunk 1]
```diff
@@ -40,6 +40,16 @@ jobs:
     timeout-minutes: 60
     steps:
       - uses: actions/checkout@v2
+      - name: Get yarn cache directory path
+        id: global-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
```
Comment (mutahhir) [inline .github/workflows/examples.yml:45]: Nit: This is already getting deprecated now, should we find an alternative?

https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs

```suggestion
        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
```

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -37,12 +37,12 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
+        id: global-cache-dir-path
         run: echo "::set-output name=dir::$(yarn cache dir)"
```
Comment (mutahhir) [inline .github/workflows/integration.yml:41]: While we're here, maybe convert these too?

**Review by mutahhir — APPROVED** [review]
I don't have a detailed understanding of how the github action caching works, but this looks like a really nice improvement overall. Just some minor nits if we can improve while we're here.

File: .github/workflows/integration.yml (hunk 2 above)
Comment (DanielMSchmidt) [reply .github/workflows/integration.yml:41]: I'm doing this in another PR, so I'll leave it for now

**Pull request merged**